File: December - 11/SakthiDharan01_counting_prime_numbers.java
import java.util.*;

public class SakthiDharan01_counting_prime_numbers {
    private static int countPrimes(int n) {
        if (n <= 2) return 0;
        boolean[] isPrime = new boolean[n];
        Arrays.fill(isPrime, true);
        isPrime[0] = isPrime[1] = false;
        for (int p = 2; p * p < n; p++) {
            if (isPrime[p]) {
                for (int mult = p * p; mult < n; mult += p) {
                    isPrime[mult] = false;
                }
            }
        }
        int count = 0;
        for (boolean b : isPrime) if (b) count++;
        return count;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt()) return;
        int n = sc.nextInt();
        System.out.println("The count of prime numbers less than " + n + " is: " + countPrimes(n));
        sc.close();
    }
}

File: December - 11/SakthiDharan01_counting_prime_numbers.py
import sys

def count_primes(n):
    if n <= 2:
        return 0
    sieve = [True] * n
    sieve[0] = sieve[1] = False
    p = 2
    while p * p < n:
        if sieve[p]:
            step = p
            start = p * p
            sieve[start:n:step] = [False] * len(range(start, n, step))
        p += 1
    return sum(sieve)

def solve():
    data = sys.stdin.read().strip().split()
    if not data:
        return
    n = int(data[0])
    print(f"The count of prime numbers less than {n} is: {count_primes(n)}")

if __name__ == "__main__":
    solve()
